nanoFramework.Json/JsonConvert.cs

```diff
@@ -442,23 +475,56 @@ private static object PopulateObject(JsonToken rootToken, Type rootType, string
 
                                         if (value.Value.GetType() != memberPropGetMethod.ReturnType)
                                         {
-                                            if (memberPropGetMethod.ReturnType.Name.Contains("Int16"))
+                                            if (memberPropGetMethod.ReturnType.Name.Contains("UInt16"))
+                                            {
+                                                memberValueArrayList.Add(Convert.ToUInt16(value.Value.ToString()));
+                                            }
+                                            else if (memberPropGetMethod.ReturnType.Name.Contains("Int16"))
                                             {
                                                 memberValueArrayList.Add(Convert.ToInt16(value.Value.ToString()));
                                             }
+                                            else if (memberPropGetMethod.ReturnType.Name.Contains("UInt32"))
+                                            {
+                                                memberValueArrayList.Add(Convert.ToUInt32(value.Value.ToString()));
+                                            }
+                                            else if (memberPropGetMethod.ReturnType.Name.Contains("Int32"))
+                                            {
+                                                memberValueArrayList.Add(Convert.ToInt32(value.Value.ToString()));
+                                            }
+                                            else if (memberPropGetMethod.ReturnType.Name.Contains("UInt64"))
+                                            {
+                                                memberValueArrayList.Add(Convert.ToUInt64(value.Value.ToString()));
+                                            }
+                                            else if (memberPropGetMethod.ReturnType.Name.Contains("Int64"))
+                                            {
+                                                memberValueArrayList.Add(Convert.ToInt64(value.Value.ToString()));
+                                            }
+                                            else if (memberPropGetMethod.ReturnType.Name.Contains("SByte"))
+                                            {
+                                                memberValueArrayList.Add(Convert.ToSByte(value.Value.ToString()));
+                                            }
                                             else if (memberPropGetMethod.ReturnType.Name.Contains("Byte"))
                                             {
-                                                memberValueArrayList.Add(Convert.ToByte(value.Value.ToString()));
+                                                memberValueArrayList.Add(Convert.ToSByte(value.Value.ToString()));
```

Comment:
ToByte !

Reply:
Thanks! Found it as well as it broke another test ;-)